src/services/validationService.ts

```diff
@@ -0,0 +1,164 @@
+import {
+  ICommonRules,
+  IRules,
+  IValidationResponse,
+  INativeQuery,
+  IDataSourceRules,
+  IQueryTypeRules,
+  IQuery,
+  ISqlQuery,
+} from "../models";
+import constants from "../resources/constants.json";
+import { isEmpty, isUndefined } from "lodash";
+import moment from "moment";
+import fs from "fs";
+import Ajv from "ajv";
+
+export class ValidationService {
+  public static validateNativeQuery(
+    query: INativeQuery,
+    commonLimits: ICommonRules,
+    dataSourceLimits: IDataSourceRules
+  ): IValidationResponse {
+    let nativeQuerySchema = JSON.parse(
+      fs.readFileSync(
+        process.cwd() + "/src/configs/nativeQuerySchema.json",
+        "utf8"
+      )
+    );
+    let ajv = new Ajv();
+    let isValidQuery = ajv.validate(
+      nativeQuerySchema,
```

Comment:
don't write everything in the singel function. 

example - 
function requestValidaion ()
function SqlQueryValidation()
function nativeQueryValidation()
function templateQueryValidation()
so it will easy to maintain the code in the later days.